File: dsa-problems/Probabilistic/solutions/javascript/PRB-016-permutation-cycle-structure.js
const readline = require("readline");

function cycleExpectations(n, k) {
  const expectedCyclesK = 1.0 / k;
  const expectedLongest = n * 0.624330;
  return [expectedCyclesK, expectedLongest];
}

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let data = [];
rl.on("line", (line) => data.push(...line.trim().split(/\s+/)));
rl.on("close", () => {
  if (data.length === 0) return;
  const n = parseInt(data[0], 10);
  const k = parseInt(data[1], 10);
  const res = cycleExpectations(n, k);
  console.log(res[0].toFixed(6) + " " + res[1].toFixed(6));
});

File: dsa-problems/Probabilistic/solutions/python/PRB-016-permutation-cycle-structure.py
import sys

def cycle_expectations(n: int, k: int):
    expected_cycles_k = 1.0 / k
    expected_longest = n * 0.624330
    return expected_cycles_k, expected_longest

def main():
    input = sys.stdin.read
    data = input().split()
    if not data:
        return
    n = int(data[0])
    k = int(data[1])
    
    res = cycle_expectations(n, k)
    print(f"{res[0]:.6f} {res[1]:.6f}")

if __name__ == "__main__":
    main()
